Storage/ThirdParty/SwiftData.swift

```diff
@@ -484,6 +494,24 @@ public class SQLiteDBConnection {
         return cursor
     }
 
+    func writeCorruptionInfoForDBNamed(dbFilename: String, toLogger logger: XCGLogger) {
+        logger.error("Corrupt DB Detected! - DB Filenamed: \(dbFilename)")
```

Comment:
I'd like to add a simple thread-safe flag to make sure we only do this once per DB. No sense spamming this full dump to disk six or seven times every time we hit it.